source/common/ssl/tls_certificate_secret_impl.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/api/v2/auth/cert.pb.h"
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Ssl {
+
+class TlsCertificateSecretImpl : public Secret::TlsCertificateSecret {
```

Comment:
I believe I mentioned this before, but this should be called `TlsCertificateConfigImpl`, since there is nothing Secret-specific about this, and both old-style certificates and SDS certificates could be represented by this object.

Basically, this should be materialized version of the [TlsCertificate](https://github.com/envoyproxy/envoy/blob/09c5d356684194f8b06df1a6b62f27a1a9cb1d69/api/envoy/api/v2/auth/cert.proto#L69) from the proto.

Reply:
Renamed back to TlsCertificateConfigImpl